Update Squire and implement hacks required for Mathjax SVG.

matthewborden opened the pull request:
This will update Squire from upstream and implement a hack to test if any SVG or Script elements are being deleted and delete their parent instead of only the element. This is because the backspace in Squire will delete the previous element to the current element, when an SVG is present this will be a use or g element not the SVG element and this will mean each number or the equation will be deleted instead of the entire equation. SVG has no support for `contentEdtiable="false"` which Squire would execute the same behaviour. We do the same for script tags, so that we can ensure that when the script tag containing the content of the equation is deleted the visual representation of the equation is not left in the DOM.

Here is the main change:

``` javascript
if (/svg|use|g|SCRIPT/.test(previous.nodeName) 
         && current.previousSibling 
         && current.previousSibling 
         && !current.previousSibling.contenteditable) {
                  detach(current.previousSibling);
                  return;
        } else {
             detach( previous );
             return;
       }
}
```

neilj committed:
Fire "select" event whenever selected range is not collapsed

Fixes #50.

xionglun committed:
fix typo: ndoes -> nodes

neilj committed:
Merge pull request #52 from xionglun/master

fix typo: ndoes -> nodes

neilj committed:
Use default block settings when fixCursor called on body

neilj committed:
Only maintain LI numbering when splitting OL if in BLOCKQUOTE.

Fixes #53

neilj committed:
Check for redundant WS when ensuring bottom line.

Fixes #55.

Commit:
Hack Squire to delete the entire block of SVG rather than each element.
This is a hack to make Squire delete the entire block of SVG when it encounters any SVG element.

Commit:
Use uppercase spelling of Script, to identify script tags.

CubedEye commented on build/squire-raw.js:
```diff
@@ -2912,8 +2919,15 @@ var keyHandlers = {
                 if ( !previous.isContentEditable ) {
-                    detach( previous );
-                    return;
+                        if (/svg|use|g|SCRIPT/.test(previous.nodeName) 
```
Isn't is better to just test these with case insensitvity? Especially when XML elements' `nodeName` is in the case it's created as (ie. are all our browsers/MathJax working together to always produce lowercase svg elements?)

CubedEye commented on source/Editor.js:
```diff
@@ -1817,8 +1829,15 @@ var keyHandlers = {
                 if ( !previous.isContentEditable ) {
-                    detach( previous );
-                    return;
+                        if (/svg|use|g|SCRIPT/.test(previous.nodeName) 
```
Isn't is better to just test these with case insensitvity? Especially when XML elements' `nodeName` is in the case it's created as (ie. are all our browsers/MathJax working together to always produce lowercase svg elements?)

Commit:
Use case insensitive regex for element comparison.
There are too many inconsitencies in browser naming of elements for use to assume that elements are lower case. Blame it on the 90's.
